File: samples/bot-task-module/python/src/models/ui_settings.py
# Copyright (c) Microsoft Corp. All rights reserved.
# Licensed under the MIT License.


class UISettings:
    """
    A class to hold UI settings for Task Module configuration.
    """

    def __init__(self, width: int, height: int, title: str, id: str, button_title: str):
        """
        Initialize UI settings.
        
        :param width: Width of the task module dialog.
        :param height: Height of the task module dialog.
        :param title: Title of the task module dialog.
        :param id: Identifier for the task module.
        :param button_title: Title for the button that invokes this task module.
        """
        self.width = width
        self.height = height
        self.title = title
        self.id = id
        self.button_title = button_title

File: samples/bot-task-module/nodejs/models/uisettings.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

/**
 * UISettings defines the UI settings for task modules in Microsoft Teams.
 */
class UISettings {
    /**
     * Creates an instance of UISettings.
     * @param {number} width - The width of the task module.
     * @param {number} height - The height of the task module.
     * @param {string} title - The title of the task module.
     * @param {string} id - The ID of the task module.
     * @param {string} buttonTitle - The button title for the task module.
     */
    constructor(width, height, title, id, buttonTitle) {
        this.width = width;
        this.height = height;
        this.title = title;
        this.id = id;
        this.buttonTitle = buttonTitle;
    }
}

module.exports.UISettings = UISettings;
